File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file157.f90
subroutine rep_nrm(u,n1,n2,n3,title,kk)



use mg_data
implicit none

integer n1, n2, n3, kk
double precision u(n1,n2,n3)
character*8 title

double precision rnm2, rnmu


call norm2u3(u,n1,n2,n3,rnm2,rnmu,nx(kk),ny(kk),nz(kk))
write(*,7)kk,title,rnm2,rnmu
7    format(' Level',i2,' in ',a8,': norms =',D21.14,D21.14)

return
end

File: data/gpt-4o/readable_format_HPEC/cpp/file157.cpp
#include <stdio.h>
#include <cmath>
void norm2u3(double u[100][100][100], int n1, int n2, int n3, double &rnm2, double &rnmu,
              int nx, int ny, int nz);
void rep_nrm(double u[100][100][100], int n1, int n2, int n3, char* title, int kk)
{
    double rnm2, rnmu;
    norm2u3(u,n1,n2,n3,rnm2,rnmu,nx,ny,nz);
    printf(" Level %d in %s: norms = %.14e %.14e
", kk, title, rnm2, rnmu);
}
